inst/@sym/private/op_helper.m

```diff
@@ -44,19 +43,20 @@
 
   % note: cmd is already cell array, hence [ concatenates with it
   cmd = [ cmd
-          '(x, y) = _ins'
-          'if x.is_Matrix and y.is_Matrix:'
-          '    assert x.shape == y.shape'
-          '    A = sp.Matrix(x.shape[0], x.shape[1],'
-          '        lambda i,j: _op(x[i,j], y[i,j]))'
-          '    return A,'
-          'elif x.is_Matrix:'
-          '    return x.applyfunc(lambda a: _op(a, y)),'
-          'elif y.is_Matrix:'
-          '    return y.applyfunc(lambda a: _op(x, a)),'
-          'else:'
-          '    return _op(x, y),' ];
+          'l = (1, 1)'
+          'for i in range(len(_ins)):'
+          '    if isinstance(_ins[i], MatrixBase):'
+          '        if _ins[i].shape != (1, 1):'
+          '            l = _ins[i].shape'
+          '        else:'
+          '            assert l == _ins[i].shape, "Matrix must have equal sizes"'
+          '    else:'
+          '        _ins[i] = Matrix([_ins[i]])'
```

Comment:
I think you should first convert it to matrix, then check the size is either same or (1,1).